agent/eventhandler/task_handler.go

```diff
@@ -347,11 +348,21 @@ func (taskEvents *taskSendableEvents) submitFirstEvent(handler *TaskHandler, bac
 	} else if event.taskShouldBeSent() {
 		if err := event.send(sendTaskStatusToECS, setTaskChangeSent, "task",
 			handler.client, eventToSubmit, handler.stateSaver, backoff, taskEvents); err != nil {
+			if isInvalidParameterException(err) {
+				// Remove the event when its parameters are invalid to reduce the redundant API call
+				seelog.Warnf("TaskHandler: Event is sent with invalid parameters; just removing: %s", event.toString())
+				taskEvents.events.Remove(eventToSubmit)
+			}
 			return false, err
 		}
 	} else if event.taskAttachmentShouldBeSent() {
 		if err := event.send(sendTaskStatusToECS, setTaskAttachmentSent, "task attachment",
 			handler.client, eventToSubmit, handler.stateSaver, backoff, taskEvents); err != nil {
+			if isInvalidParameterException(err) {
```

Comment:
Could you just use `if utils.IsErrorCodeEqual(err, ecs.ErrCodeInvalidParameterException)` here since the method `isInvalidParameterException` is not adding much value?

Reply:
Instead the below two statements can be moved to a method called `handleInvalidParamException` or something similar.

Reply:
oh I see what you mean.